src/main/java/redis/clients/jedis/JedisSentinelPool.java

```diff
@@ -139,9 +140,7 @@ private HostAndPort initSentinels(Set<String> sentinels, final String masterName
 
       log.fine("Connecting to Sentinel " + hap);
 
-      Jedis jedis = null;
-      try {
-        jedis = new Jedis(hap.getHost(), hap.getPort());
+      try (Jedis jedis = new Jedis(hap.getHost(), hap.getPort())) {
```

Comment:
I understand your point and I believe it's valid. What I'm saying is that makes more difficult to backport stuff to Jedis 2.x as we'll have to do manual merging because compilation will fail in those branches.